### PR TITLE
[MAINTENANCE] Upgrade to modern Python syntax

### PR DESCRIPTION
**Pull request by cclauss**
This pull request was created with the command [`pyupgrade --py36-plus **/*.py`](https://github.com/asottile/pyupgrade#readme) followed by `black .`

Python f-strings simplify the code and [should speed up execution](https://www.scivision.dev/python-f-string-speed). 

Please annotate your PR title to describe what the PR does, then give a brief bulleted description of your PR below. PR titles should begin with [BUGFIX], [FEATURE],  [DOCS], or [MAINTENANCE]. If a new feature introduces breaking changes for the Great Expectations API or configuration files, please also add [BREAKING]. You can read about the tags in our [contributor checklist](https://docs.greatexpectations.io/en/latest/contributing/contribution_checklist.html).

Changes proposed in this pull request:
-
-
-

After submitting your PR, CI checks will run and @tiny-tim-bot will check for your CLA signature.

For a PR with nontrivial changes, we review with both design-centric and code-centric lenses.

In a design review, we aim to ensure that the PR is consistent with our relationship to the open source community, with our software architecture and abstractions, and with our users' needs and expectations. That review often starts well before a PR, for example in github issues or slack, so please link to relevant conversations in notes below to help reviewers understand and approve your PR more quickly (e.g. `closes #123`).

Previous Design Review notes:
-
-
-

### Definition of Done
Please delete options that are not relevant.

- [ ] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/en/latest/contributing/style_guide.html?highlight=style%20guide)
- [ ] I have performed a [self-review](https://docs.greatexpectations.io/en/latest/contributing/contribution_checklist.html?highlight=checklist) of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added [unit tests](https://docs.greatexpectations.io/en/latest/contributing/testing.html#contributing-testing-writing-unit-tests) where applicable and made sure that new and existing tests are passing.
- [ ] I have run any local integration tests and made sure that nothing is broken.

Thank you for submitting!

### CODE REVIEW

File: great_expectations/datasource/data_connector/util.py
Comment (cdkini) [inline great_expectations/datasource/data_connector/util.py:199]: Is it necessary to make this a raw string? My familiarity with anything outside of `f-strings` is limited but I'm not sure this adds anything.

File: great_expectations/data_context/store/database_store_backend.py
Comment (cdkini) [inline great_expectations/data_context/store/database_store_backend.py:251]: We see a few cases of this in the proposed changes; is there a case where converting this from a list to a generator causes us a problem? I would think it would only provide benefits (reduced memory usage) but I'm not confident.

**Review by cdkini — COMMENTED** [review]
I think these changes are quite helpful in pushing our codebase forward to a more modern version of Python but I'd like to review a few things before I sign off.

1. Is it always safe to replace a list with a generator in the cases you've fixed?
  - I think yes (especially with `sorted`).
2. Are `f-strings` and `.format()` identical? Is there any risk in treating them as exactly the same thing? 
  - I think yes as long as the Python parser follows the same order of operations.
3. When should we utilize less common string modifiers such as `b-strings` and `r-strings`? Does it make sense to use them as they've been used here?
  - I think we should avoid using them since they provide little value in the cases brought up.

Let me gain some understanding around these questions before I move forward.

File: great_expectations/datasource/data_connector/util.py
Comment (cclauss) [reply great_expectations/datasource/data_connector/util.py:199]: Pyupgrade uses a raw string here because of the [invalid \d escape sequences](https://github.com/asottile/pyupgrade#invalid-escape-sequences) on line 204.  I believe that pytest will also raise warnings about these

File: great_expectations/data_context/store/database_store_backend.py
Comment (cclauss) [reply great_expectations/data_context/store/database_store_backend.py:251]: We are calling `and_()` with these *args so I do not see how *list comprehension vs. *generator expression will make a difference in populating the parameters other that the efficiency that you site.  If you see something that I do not then an issue to https://github.com/asottile/pyupgrade might be the best approach.

**Comment by cdkini** [comment]
@cclauss I definitely agree with your comments and overall approach so thank you for yet another PR 🚀  

However, due to the number of changes, I think getting another set of eyes to take a quick look would be of value. We'll definitely resolve this and have it merged in soon enough. Appreciate your patience.

**Review by talagluck — APPROVED** [review]
Thanks for the contribution, @cclauss - we've reviewed this PR again, and it looks great!

**Pull request merged**
